### ISSUE TITLE
Close time of campaign needs clarification

### ISSUE THREAD

**Issue by MichaelNMaggs** [hatnote/montage#62]
The setup instructions at https://commons.wikimedia.org/wiki/Commons:Montage#Instructions_for_coordinators currently say:

- _Make sure [the close time] is not the actual end time of your campaign. If you go with the actual end time, any photo that has been edited (even by users other than the initial uploader) after the end time of your campaign will be excluded from the jury process and this is not what you want._

That looks like a temporary fix. The program should be looking at the initial upload time of each file, not the last edited time.  As set up at present, files uploaded after the contest has ended, but before the time entered in this field, will incorrectly be included.

**Comment by leilaz** [comment]
@MichaelNMaggs this is not actually as clear cut and easy to implement. First off, we looked into this last year and because of the way Commons data is stored, it would be computationally expensive to retrieve the first version of the file. Also, sometimes the uploader themselves improve their file over time (and within the campaign time) so their first upload is not the best upload. Lastly, some countries allow the uploader to edit images even after the end of the contest without guarantee that the latest version will be considered. All in all, it seems the best working solution is to retrieve the last image for now.

Stephen and Mahmoud may want to speak more to this, as we discussed this extensively a couple of times.

**Comment by MichaelNMaggs** [comment]
I agree that retrieving the last image makes sense, but the initial upload date should never be after the end date of the contest.  The initial version does not need to be retrieved, only the upload date checked. Maybe the computational load could be reduced by checking the initial upload date only for those images that have a most recent edit after the contest end date.  Any files that have a most recent and and initial upload date after the contest end date should be disqualified.

**Comment by lgelauff** [comment]
(see also #155 )
